### ISSUE TITLE
"Upload failed: Could not complete operation. Server unavailable."

### ISSUE THREAD

**Issue by despens** [nextcloud/android#2285]
### Actual behaviour
Photos are not uploaded to the Nextcloud instance via Auto Upload. Instead, a notification stating "Upload failed: Could not complete operation. Server unavailable." is generated each time an upload is attempted.
Ironically I am using the Nextcloud android app to upload the app's log to the exact Nextcloud instance it doesn't upload the photos to. :smile: 

### Expected behaviour
With *Auto upload* being activated, photos should be uploaded to Nextcloud instance.
 
### Steps to reproduce
1. Activate Auto upload for camera folder

### Environment data
Android version: 6.0.1

Device model: STV100-1

Stock or customized system: stock

Nextcloud app version: 3.0.2

Nextcloud server version: 13.0.0

#### Web server error log
Cannot locate relevant lines :disappointed: 

#### Nextcloud log (data/nextcloud.log)

level | app | message
-- | -- | --
Debug | webdav | Sabre\DAV\Exception\NotAuthenticated: No  public access to this resource., No 'Authorization: Basic' header  found. Either the client didn't send one, or the server is  misconfigured, No 'Authorization: Bearer' header found. Either the  client didn't send one, or the server is mis-configured, No  'Authorization: Basic' header found. Either the client didn't send one,  or the server is misconfigured[internal function] Sabre\DAV\Auth\Plugin->beforeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))/var/www/nextcloud/3rdparty/sabre/event/lib/EventEmitterTrait.php - line 105:  call_user_func_array(Array, Array)/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 466:  Sabre\Event\EventEmitter->emit('beforeMethod', Array)/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 254:  Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))/var/www/nextcloud/apps/dav/lib/Server.php - line 283:  Sabre\DAV\Server->exec()/var/www/nextcloud/apps/dav/appinfo/v2/remote.php - line 35:  OCA\DAV\Server->exec()/var/www/nextcloud/remote.php - line 164:  require_once('/var/www/nextcl...'){main}

#### App log

```
2018/03/04 15:41:21
UploadsStorageManager : updateUpload returns with: 1 for file: /storage/emulated/0/DCIM/Camera/IMG_20180303_1012392.jpg

2018/03/04 15:41:21
UploadsStorageManager : notifyObserversNow

2018/03/04 15:41:22
FileUploader : Error uploading Exception : [com.owncloud.android.operations.UploadFileOperation.run(UploadFileOperation.java:374), com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:145), com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:149), com.owncloud.android.operations.common.SyncOperation.execute(SyncOperation.java:93), com.owncloud.android.files.services.FileUploader.uploadFile(FileUploader.java:1090), com.owncloud.android.files.services.FileUploader$ServiceHandler.handleMessage(FileUploader.java:1023), android.os.Handler.dispatchMessage(Handler.java:102), android.os.Looper.loop(Looper.java:148), android.os.HandlerThread.run(HandlerThread.java:61)]

2018/03/04 15:41:22
UploadsStorageManager : updateDatabaseUploadResult uploadResult: com.owncloud.android.lib.common.operations.RemoteOperationResult@2fe14f1 upload: com.owncloud.android.operations.UploadFileOperation@3f688d6

2018/03/04 15:41:22
UploadsStorageManager : Updating /storage/emulated/0/DCIM/Camera/IMG_20180303_1012392.jpg with status:UPLOAD_FAILED and result:NETWORK_CONNECTION (old:/storage/emulated/0/DCIM/Camera/IMG_20180303_1012392.jpg status:UPLOAD_IN_PROGRESS result:-1)
```

**Comment by tribut** [comment]
This is probably a duplicate of #2278, possible workaround in https://github.com/nextcloud/android/issues/2278#issuecomment-370229046

**Comment by enoch85** [comment]
Same here @mario :) Looking forward to a fix.

**Comment by despens** [comment]
I did create the new folder `/InstantUploads/2018/03` as suggested in #2278, but my photos are still not uploading with the same error.

**Comment by tribut** [comment]
Did you try and navigate to the folder in the Nextcloud App? If its the same bug, the app will not contact the server at all unless it "knows" the folder exists.

**Comment by xXSTrikeXx** [comment]
@despens Can you try latest version 3.0.3?

**Comment by enoch85** [comment]
This is fixed in 3.0.3 just tested myself.

Please reopen if you think I'm wrong.
